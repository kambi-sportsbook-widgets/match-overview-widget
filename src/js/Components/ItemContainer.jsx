import React, { PropTypes } from 'react';
import styles from './ItemContainer.scss';

/**
 * Returns DOM element's width (in pixels)
 * @param {HTMLElement} el DOM element
 * @returns number|null
 */
const getWidth = function(el) {
   return el ? el.offsetWidth : null;
};

const ItemContainer = ({ children, selected, onClick, onWidth }) => (
   // onClick is not wired in order to disable scroll on click

   <div
      className={styles.item}
      ref={onWidth ? el => onWidth(getWidth(el)) : undefined}
   >
      {children}
   </div>
);

ItemContainer.propTypes = {
   /**
    * On click handler
    */
   onClick: PropTypes.func,

   /**
    * Is this tab currently selected?
    */
   selected: PropTypes.bool,

   /**
    * Tab contents
    */
   children: PropTypes.node,

   /**
    * Called when tab width is known
    */
   onWidth: PropTypes.func
};

export default ItemContainer;
