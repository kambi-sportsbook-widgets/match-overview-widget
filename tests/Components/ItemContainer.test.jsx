/* eslint-env jest */
import React from 'react';
import ItemContainer from '../../src/js/Components/ItemContainer';
import ReactTestUtils from 'react-addons-test-utils';
import { shallow, mount } from 'enzyme';

let renderer;

describe('ItemContainer view', () => {

   beforeEach(() => {
      renderer = ReactTestUtils.createRenderer();
   });

   it('renders correctly with default props', () => {
      expect(renderer.render(
         <ItemContainer />
      )).toMatchSnapshot();
   });

   it('renders correctly selected state', () => {
      expect(renderer.render(
         <ItemContainer selected={true} />
      )).toMatchSnapshot();
   });

   it('renders correctly with one child element', () => {
      expect(renderer.render(
         <ItemContainer>One</ItemContainer>
      )).toMatchSnapshot();
   });

   it('renders correctly with many child elements', () => {
      expect(renderer.render(
         <ItemContainer>
            One
            Two
            <ul>
               <li>Three</li>
            </ul>
         </ItemContainer>
      )).toMatchSnapshot();
   });

});

describe('ItemContainer interface', () => {

   it('doesn\'t react to click events', () => {
      const onClickMock = jest.fn();

      const wrapper = shallow(
         <ItemContainer onClick={onClickMock} />
      );

      expect(onClickMock).not.toHaveBeenCalled();

      wrapper.first().simulate('click');

      expect(onClickMock).toHaveBeenCalledTimes(0);
   });

   it('calls onWidth after mounting', () => {
      const onWidthMock = jest.fn();

      mount(<ItemContainer onWidth={onWidthMock} />);

      expect(onWidthMock).toHaveBeenCalledTimes(1);
   });

});
