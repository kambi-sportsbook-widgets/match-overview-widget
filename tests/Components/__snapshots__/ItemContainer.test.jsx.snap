// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemContainer view renders correctly selected state 1`] = `
<div
  className="item"
/>
`;

exports[`ItemContainer view renders correctly with default props 1`] = `
<div
  className="item"
/>
`;

exports[`ItemContainer view renders correctly with many child elements 1`] = `
<div
  className="item"
>
  One Two
  <ul>
    <li>
      Three
    </li>
  </ul>
</div>
`;

exports[`ItemContainer view renders correctly with one child element 1`] = `
<div
  className="item"
>
  One
</div>
`;
